components/builder-api-client/src/lib.rs

```diff
@@ -0,0 +1,160 @@
+// Copyright (c) 2016-2017 Chef Software Inc. and/or applicable contributors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#![cfg_attr(feature="clippy", feature(plugin))]
+#![cfg_attr(feature="clippy", plugin(clippy))]
+
+extern crate habitat_http_client as hab_http;
+extern crate habitat_core as hab_core;
+extern crate hyper;
+extern crate hyper_openssl;
+#[macro_use]
+extern crate log;
+extern crate regex;
+extern crate serde;
+#[macro_use]
+extern crate serde_derive;
+#[macro_use]
+extern crate serde_json;
+extern crate url;
+
+pub mod error;
+pub use error::{Error, Result};
+
+use std::io::Read;
+use std::path::Path;
+
+use hab_core::package::PackageIdent;
+use hab_http::ApiClient;
+use hyper::client::{IntoUrl, Response, RequestBuilder};
+use hyper::header::{Accept, Authorization, Bearer, ContentType};
+use hyper::status::StatusCode;
+use regex::Regex;
+
+pub struct Client(ApiClient);
+
+#[derive(Deserialize)]
+pub struct ReverseDependencies {
+    pub origin: String,
+    pub name: String,
+    pub rdeps: Vec<String>,
+}
+
+impl Client {
+    pub fn new<U>(
+        depot_url: U,
+        product: &str,
+        version: &str,
+        fs_root_path: Option<&Path>,
+    ) -> Result<Self>
+    where
+        U: IntoUrl,
+    {
+        let re = Regex::new(r"depot/?$").unwrap();
+        let mut url = depot_url.into_url().map_err(Error::URL)?;
+
+        if re.is_match(url.path()) {
+            let mut psm = url.path_segments_mut().unwrap();
+            psm.pop();
+        }
+
+        let api_url = url.as_str();
+
```

Comment:
Because `builder-api`'s base URL is `/v1` and `builder-depot`'s base URL is `/v1/depot`.  All of this exists for legacy reasons, but since everything is already oriented around the depot URL, including things like environment variables, it seemed silly to add another variable for everyone to track, especially when there are plans to consolidate all of this at some point.  So instead, `builder-api-client` just accepts the depot URL as a parameter and converts it to the api URL by popping `depot` off the end.